crates/macro-support/src/parser.rs

```diff
@@ -377,11 +377,26 @@ impl<'a> ConvertToAst<BindgenAttrs> for &'a mut syn::ItemStruct {
             .unwrap_or(self.ident.to_string());
         let is_inspectable = attrs.inspectable().is_some();
         let getter_with_clone = attrs.getter_with_clone().is_some();
+        let transparent = if self.attrs.iter().any(|attr| match attr.parse_meta() {
+            Ok(Meta::List(MetaList { path, nested, .. })) => {
+                path.is_ident("repr")
+                    && nested.iter().any(|nested| match nested {
+                        NestedMeta::Meta(Meta::Path(path)) => path.is_ident("transparent"),
+                        _ => false,
+                    })
+            }
+            _ => false,
+        }) {
+            self.fields.iter().next().map(|field| field.ty.clone())
```

Comment:
Given that transparent structures can have multiple fields so long as only one has nonzero size, I don't think that this is the correct algorithm to find the type that we're being transparent over?

Reply:
Yeah I don't think this is either. But I don't think there's a way to know if a given field has a non-zero size.

Maybe we will have to force the user to implement a trait like `TransparentBind` so that we can refer to the type as `TransparentBind::NonZeroSizedFieldType` (with a more acceptable name of course haha)

But, just like the `From` implementation, I feel like this is a lot of extra work to get `#[repr(transparent)]` to compile when targeting the Web vs other platforms.

Reply:
That's true yeah, and I also feel like this is perhaps getting a bit too complicated. If this can't be done easily then I think it would be best to shelf the idea for now perhaps?